inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -416,9 +459,131 @@ export class CanvasGraphService {
      */
     private _removeEventListeners(canvas: HTMLCanvasElement) {
         canvas.removeEventListener("wheel", this._handleZoom);
+        canvas.removeEventListener("mousemove", this._handleDataHover);
         canvas.removeEventListener("mousedown", this._handlePanStart);
+        canvas.removeEventListener("mouseleave", this._handleStopHover);
         canvas.ownerDocument.removeEventListener("mouseup", this._handlePanStop);
     }
+    
+    /**
+     * Handles what to do when we are hovering over the canvas and not panning.
+     * 
+     * @param event A reference to the event to be handled. 
+     */
+    private _handleDataHover = (event: MouseEvent) => {
+        if (this._panPosition) {
+            // we don't want to do anything if we are in the middle of panning
+            return;
+        }
+
+        this._hoverPosition = event.clientX;
+
+        // then draw the tooltip.
+        this._drawTooltip(this._hoverPosition, this._drawableArea);
+    }
+
+    /**
+     * Handles what to do when we stop hovering over the canvas.
+     */
+    private _handleStopHover = () => {
+        this._hoverPosition = null;
+    }
+
+    /**
+     * Draws the tooltip given the area it is allowed to draw in and the current pixel position.
+     * 
+     * @param pixel the position of the mouse cursor in pixels. 
+     * @param drawableArea  the available area we can draw in.
+     */
+    private _drawTooltip(pixel: number | null, drawableArea: IGraphDrawableArea) {
+        const {_ctx : ctx} = this;

```

Comment:
I wonder how slow the function is and if it should debounce a bit looking into the rest of the code ?

Reply:
Seems to be ~0.13ms: I could possibly debounce though if you feel it is necessary

Reply:
Okay so I debounced the drawing part inside of the event handler I noticed that the actual event itself needed to update the mouse position whenever, but the expensive bit inside the mousemove has been debounced. This makes the mouse move event a lot less performance heavy!